icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -2127,10 +2108,11 @@ void SimpleDateFormat::adoptNumberFormat(NumberFormat *formatToAdopt) {
         fSharedNumberFormatters = NULL;
     }
 
-    // Also re-compute the fast formatters.
+    // Also re-compute the fast formatter.
     UErrorCode localStatus = U_ZERO_ERROR;
-    freeFastNumberFormatters();
-    initFastNumberFormatters(localStatus);
+    delete fSimpleNumberFormatter;
+    fSimpleNumberFormatter = nullptr;
+    initSimpleNumberFormatter(fLocale, localStatus);
```

Comment:
This implementation of adoptNumberFormat won't work anymore since we're just reinitializing with the locale. I wonder what we should do here? Perhaps set an `adopted` flag that disables the simple formatter, or somehow extract symbol/locale info from the NumberFormat?